pkg/provisioning/service.go

```diff
@@ -208,27 +213,6 @@ func (s *Service) getYamlFiles(ctx context.Context, path string) ([]string, erro
 	return files, nil
 }
 
-func (s *Service) resolveSymLink(basePath, symlink string) (string, error) {
-	fullPath := filepath.Join(basePath, symlink)
-	resolvedPath, err := os.Readlink(fullPath)
-	if err != nil {
-		return "", cerrors.Errorf("could not read symlink: %w", err)
-	}
-
-	// If symlink path is relative, make it absolute
-	if !filepath.IsAbs(resolvedPath) {
-		resolvedPath = filepath.Join(basePath, resolvedPath)
-	}
-
-	// Check if the resolved path exists and is a regular file
-	_, err = os.Stat(resolvedPath)
-	if err != nil {
-		return "", cerrors.Errorf("could not stat symlink: %w", err)
-	}
-
-	return resolvedPath, nil
-}
-
 func (s *Service) isYamlFile(path string) bool {
```

Comment:
How about we pass the whole `dirEntry` in here and use `dirEntry.IsDir()` to determine if it's a file or a dir?